Client/Frontend/Toolbar+URLBar/OverlayModeManager.swift

```diff
@@ -49,20 +49,32 @@ class DefaultOverlayModeManager: OverlayModeManager {
         enterOverlayMode(pasteContent, pasted: true, search: true)
     }
 
-    func openNewTab(_ locationText: String?, url: URL?) {
-        if url == nil || url?.isFxHomeUrl ?? false {
-            enterOverlayMode(locationText, pasted: false, search: true)
-        }
+    func openNewTab(url: URL?, newTabSettings: NewTabPage) {
+        guard shouldEnterOverlay(for: url, newTabSettings: newTabSettings) else { return }
+
+        enterOverlayMode(nil, pasted: false, search: true)
     }
 
-    func finishEdition(shouldCancelLoading: Bool) {
+    func finishEditing(shouldCancelLoading: Bool) {
         leaveOverlayMode(didCancel: shouldCancelLoading)
     }
 
     func switchTab(shouldCancelLoading: Bool) {
+        guard inOverlayMode else { return }
+
         leaveOverlayMode(didCancel: shouldCancelLoading)
     }
 
+    private func shouldEnterOverlay(for url: URL?, newTabSettings: NewTabPage) -> Bool {
+        // The NewTabPage cases are weird topSites = homepage
+        // and homepage = customURL
```

Comment:
I think initially the homepage only contained top sites or sth like that. The homepage was sometimes referred with ActivityStream. 🤔 Wondering if (in another story) we should rename those newTabSettings to avoid confusion since our naming convention since that code was written.

Reply:
I think it makes sense to rename in another ticket, it confused me a lot when I was debugging. The main reason why I didn't update is because of the value saved on pref